### ISSUE TITLE
18c XE box tnsnames.ora enhancement

### ISSUE THREAD

**Issue by PaulNeumann** [oracle/vagrant-boxes#104]
There are 2 issues with the tnsnames.ora file for the Oracle Database 18c XE box that make connecting to the database from an SSH session a little harder than it is with the other database boxes.

First, unlike the other database boxes, the permissions on tnsnames.ora allow read access to the oracle user and the oinstall group, but not to others, including the vagrant user.

Permissions on tnsnames.ora for the 12cR2 box:
```
-rw-r--r--. 1 oracle oinstall 197 Feb 26 20:05 tnsnames.ora
```
Permissions on tnsnames.ora for the 18c XE box:
```
-rw-r-----. 1 oracle oinstall 395 Feb 25 18:57 tnsnames.ora
```

This means that `sqlplus <user>/<password>@XE` fails for the vagrant user with `ORA-12154: TNS:could not resolve the connect identifier specified`, even after running `oraenv`.

Second, unlike the 12c and 18c boxes, the tnsnames.ora file doesn't have an entry for the PDB. So, even after changing the permissions on tnsnames.ora, `sqlplus <user>/<password>@XEPDB1` fails with `ORA-12154`.

Adding the following to `install.sh` right after running `oracle-xe-18c configure` would remedy both issues, and would make the box easier to use, especially for users who may not be familiar with Easy Connect syntax:
```bash
chmod o+r /opt/oracle/product/18c/dbhomeXE/network/admin/tnsnames.ora

# add tnsnames.ora entry for PDB
echo 'XEPDB1 =
  (DESCRIPTION =
    (ADDRESS = (PROTOCOL = TCP)(HOST = localhost)(PORT = 1521))
    (CONNECT_DATA =
      (SERVER = DEDICATED)
      (SERVICE_NAME = XEPDB1)
    )
  )
' >> /opt/oracle/product/18c/dbhomeXE/network/admin/tnsnames.ora
```
If you think that this would be helpful, I'd be happy to submit a PR.

**Comment by gvenzl** [comment]
Yes @PaulNeumann, that would be great! Please feel free to submit a PR!

**Comment by PaulNeumann** [comment]
Thank you, sir. I've submitted a PR for this.

(As a side note, it looks like the 19c installer works the same way, at least for 19.2, so similar code might be helpful for the future 19c Vagrant box.)
